src/lib/classes/services/chainsService.ts

```diff
@@ -120,15 +135,62 @@ export class ChainsService {
         ]);
 
         this._cachedChains[chainID].blocks[blockIndex] = block;
-        this.save();
+
+        this.cacheBlocksChain(chainID, blockIndex, block); // Cache the block
 
         return block;
     }
 
-    /**
-     * Persist the cached blocks
-     */
-    public save() {
-        this._storageService.save("chains", this._cachedChains);
+    private cacheBlocksChain(chainID: string, blockIndex: number, block: BlockData) {
+        const cachedChainData = this._cachedBlocksOfChain[chainID] ?? { blocks: [] };
+
+        // Check if the block already exists
+        const existingBlockIndex = cachedChainData.blocks.findIndex(
+            // eslint-disable-next-line @typescript-eslint/no-shadow
+            block => block.id === blockIndex,
+        );
+
+        // If it exists, remove it
+        if (existingBlockIndex !== -1) {
+            cachedChainData.blocks.splice(existingBlockIndex, 1);
+        }
+
+        // Add the block to the cache
+        const blockCachedData: BlockCachedData = {
+            id: blockIndex,
+            block,
+        };
+
+        cachedChainData.blocks.push(blockCachedData);
+
+        // Limit the number of blocks in cache
+        const maxLength = 100;
+        if (cachedChainData.blocks.length > maxLength) {
+            // Find the block that is the furthest away from the current block
+            let maxDifference = Number.NEGATIVE_INFINITY;
+            let indexToRemove = -1;
+            for (let i = 0; i < maxLength; i++) {
+                const difference = Math.abs(blockIndex - cachedChainData.blocks[i].id);
+                if (difference > maxDifference) {
+                    maxDifference = difference;
+                    indexToRemove = i;
+                }
+            }
+            if (indexToRemove !== -1) {
+                cachedChainData.blocks.splice(indexToRemove, 1);
+            }
+        }
+        // Save the updated cached blocks
+        this._cachedBlocksOfChain[chainID] = cachedChainData;
+
+        // Save the updated cached blocks
+        this.saveCachedBlocks();
+    }
+
+    private saveCachedBlocks() {
+        this._storageService.save("chain with blocks", this._cachedBlocksOfChain);
+        if (this.cacheBlocksChain !== undefined) {
```

Comment:
I would generate a migration manager in the project to manage any necessary migration, versioning whats stored in the localStorage